Source/Core/DolphinQt/MenuBar.cpp

```diff
@@ -1452,6 +1458,16 @@ void MenuBar::SaveCode()
 
 bool MenuBar::TryLoadMapFile(const QString& path, const bool bad)
 {
+  if (!MSR.DR)
```

Comment:
IMO this check for DR should still be in LoadMap. Accessing MSR directly in UI code is kind of... ugly? Make LoadMap return an enum so you can detect if loading has failed because of translation being disabled if you want to show a different error message.

Also, this check (written as is) introduces another race condition.

Reply:
Making an enum for just that seems uglier and less clear.
How could there be a race condition if MSR.DR never changes?  All three functions calling it are using Core::RunAsCPUThread.

Reply:
How is it less clear if the check looks like `if (error == LoadMapError::TranslationDisabled)`? I'd argue that's actually clearer than checking for the relevant bit in MSR directly and it doesn't expose things that callers shouldn't have to know about.

Another reason for putting the check for MSR.DR is that it's still needed in case LoadMap is called when translation is disabled. This is unlikely to happen during a game boot but it's possible so I think it makes more sense to check it there.

I don't feel that strongly about this though, so maybe let's wait for another opinion.

> How could there be a race condition if MSR.DR never changes? All three functions calling it that were experiencing race conditions are now using Core::RunAsCPUThread.

Oh right, this is called from the CPU thread. Forget what I said about the race then.